File: src/common/py/component/component_data.py
import dataclasses

from semantic_version import Version

from .roles.component_role import ComponentRole
from ..utils import UnitID
from ..utils.config import Configuration


@dataclasses.dataclass(frozen=True, kw_only=True)
class ComponentData:
    """
    Holds general data and information about the component.
    
    Objects of this class are passed to certain parts of the core for easy access to frequently
    used component information and data, like its configuration or role.
    
    Attributes:
        comp_id: The component identifier.
        role: The component role.
        config: The configuration.
        title: The project title.
        name: The component name.
        version: The project version.
    """
    comp_id: UnitID
    role: ComponentRole
    
    config: Configuration
    
    title: str
    name: str
    version: Version

File: src/common/web/component/ComponentData.ts
import { SemVer } from "semver";

import { Configuration } from "../utils/config/Configuration";
import { UnitID } from "../utils/UnitID";

/**
 * Holds general data and information about the component.
 *
 * Objects of this class are passed to certain parts of the core for easy access to frequently
 * used component information and data.
 */
export class ComponentData {
    /**
     * @param compID - The component identifier.
     * @param config - The configuration.
     * @param title - The project title.
     * @param name - The component name.
     * @param version - The project version.
     */
    public constructor(readonly compID: UnitID, readonly config: Configuration,
                       readonly title: string, readonly name: string, readonly version: SemVer) {
    }
}
